src/condor_includes/proc.h

```diff
@@ -147,6 +147,19 @@ typedef struct JOB_ID_KEY {
 inline bool operator==( const JOB_ID_KEY a, const JOB_ID_KEY b) { return a.cluster == b.cluster && a.proc == b.proc; }
 size_t hashFunction(const JOB_ID_KEY &);
 
+// Macros used to indicate how a job was submitted to htcondor
+#define JOB_SUBMIT_METHOD_MIN 0
+#define CONDOR_SUBMIT 0
+#define DAGMAN 1
+#define PYTHON_BINDINGS 2
+#define HTC_JOB_SUBMIT 3
+#define HTC_DAG_SUBMIT 4
+#define HTC_JOBSET_SUBMIT 5
+#define USER_SET 100
+#define JOB_SUBMIT_METHOD_MAX 100
```

Comment:
give these a common prefix like METH_ or JSM_ to help avoid unexpected expansion of these macros in code that does not intend it.  bare DAGMAN seems particularly dangerous.